src/App/Fossa/Analyze.hs

```diff
@@ -125,7 +139,24 @@ analyzeMain workdir recordMode logSeverity destination project unpackArchives fi
             let effects = analyzeEffects journal
             runReplay @ReadFS (effectsReadFS effects)
               . runReplay @Exec (effectsExec effects)
-              $ analyze basedir destination project unpackArchives filters
+              $ analyze basedir destination project unpackArchives enableVSI filters
+
+-- vsiDiscoverFunc is appended to discoverFuncs during analyze.
+-- It's not added to discoverFuncs because it requires more information than other discoverFuncs.
+vsiDiscoverFunc ::
+  ( Has (Lift IO) sig m,
+    Has Diag.Diagnostics sig m,
+    Has (Lift IO) rsig run,
+    MonadIO run,
+    Has Diag.Diagnostics rsig run,
+    Has Exec rsig run
+  ) =>
```

Comment:
Let's pull out this constraint as an alias:
```hs
type TaskEffs sig m =
  ( Has (Lift IO) sig m
  , MonadIO m
  , Has ReadFS sig m
  , Has Exec sig m
  , Has Logger sig m
  , Has Diag.Diagnostics sig m
  )
```

then we can do:
```suggestion
vsiDiscoverFunc ::
  ( TaskEffs sig m,
    TaskEffs rsig run
  ) =>
```

and also change the type of `discoverFuncs` to use `TaskEffs sig m, TaskEffs rsig run` instead of the mess of a type signature it has right now

Reply:
Since `TaskEffs` asserts `Has Logger`, I can't replace the `rsig run` portion of `discoverFuncs` without getting a type error in `listTargetsMain`:
```
No instance for (Control.Effect.Sum.Member
                     Logger
                     (fused-effects-1.1.1.1:Control.Effect.Lift.Internal.Lift IO))
    arising from a use of ‘discoverFuncs’
```

I've updated the signature to:
```haskell
discoverFuncs ::
  ( TaskEffs sig m,
    Has (Lift IO) rsig run,
    Has Diag.Diagnostics rsig run,
    Has Exec rsig run,
    Has ReadFS rsig run
  ) =>
  -- | Discover functions
  [Path Abs Dir -> m [DiscoveredProject run]]
```

Do you want me to leave it like this? I'm kind of hesitant to touch anything in `listTargetsMain` since it's not necessary to change its functionality for the VSI strategy, but I'm happy to do whatever you want me to do.

Reply:
Ah whoops. Let's update the `DummyM` type alias in `App.Fossa.ListTargets`:

```hs
type DummyM = ReadFSIOC (ExecIOC (Diag.DiagnosticsC (LoggerC IO)))
```

Reply:
Ok, done!